airflow/contrib/operators/dataproc_operator.py

```diff
@@ -243,8 +243,92 @@ def __init__(self,
             )
         ), "num_workers == 0 means single node mode - no preemptibles allowed"
 
+    def _get_cluster_list_for_project(self, service):
+        result = service.projects().regions().clusters().list(
+            projectId=self.project_id,
+            region=self.region
+        ).execute()
+        return result.get('clusters', [])
+
+    def _get_cluster(self, service):
+        cluster_list = self._get_cluster_list_for_project(service)
+        cluster = [c for c in cluster_list if c['clusterName'] == self.cluster_name]
+        if cluster:
+            return cluster[0]
+        return None
+
+    def _get_cluster_state(self, service):
```

Comment:
I think this piece of code will be better suited to the hook. In addition, it is worth adding a mechanism to repeat requests
```suggestion
        ).execute(num_retries=self.num_retries)
```